### PR TITLE
Composite ELMO: first part.

### PR DESCRIPTION
**Pull request by h0nzZik**

### CODE REVIEW

File: VLSM/ELMO/ELMO.v [hunk 1]
```diff
@@ -1611,4 +1610,1030 @@ Section composition.
...
+    has_been_received_capability free_composite_elmo.
+  Proof.
+    unfold free_composite_elmo. unfold free_composite_vlsm.
+    eapply composite_has_been_received_capability.
+    { apply finite_index. }
+    intros i.
+    apply elmo_has_been_received_capability.
+  Defined.
+
+  Lemma has_receive_observation_implies_is_protocol st idx m:
```
[the remark this answers is not in the document]
Comment (h0nzZik) [reply VLSM/ELMO/ELMO.v:1669]: I actually use `protocol_trace_input_is_protocol` at the end of the proof.

**Review by palmskog — CHANGES_REQUESTED** [review]
This looks reasonable in general, but can we get a consistent style for use of blocks `{ ... }` in the proof script?

For example, I would recommend:

`{ intuition. }` consistently for one-liners

```
{
  intution.
  lia.
}
```
for multiple-liners.

**Review by traiansf — APPROVED** [review]
I'm ok with it being merged once Karl is satisfied with it.

**Review by palmskog — APPROVED** [review]
This now looks good, only thing that possiblty remains is to rebase on top of #228 if we merge that first.

**Pull request merged**